### PR TITLE
DBZ-1019 Guard against wall-clock change

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1019

### CODE REVIEW

**Comment by gunnarmorling** [comment]
I'm not quite sure I'm following. Could you expand the commit message perhaps to clarify what's going on? Thanks!

**Comment by jpechane** [comment]
@gunnarmorling The original issue reproted a negative value passed to the `sleep` method. The modified method indirectl calls `System.currentTimeMillis()`. Thos method depends on the system clock of the computer. It is thus possible that time can decrease if for exmaple NTP is syncing clock. So it is possible that in two consecutive calls is `timeSystem.currentTimeInMillis()` larger than next and then smaller which causes the exception. Hence its necessary to read the value and process it only once. The problem is similar to a non-atomic access to a variable in multithreading program.

File: debezium-core/src/main/java/io/debezium/util/Metronome.java [hunk 1]
```diff
@@ -50,8 +50,12 @@ public static Metronome sleeper(long period, TimeUnit unit, Clock timeSystem) {
 
             @Override
             public void pause() throws InterruptedException {
-                while (next > timeSystem.currentTimeInMillis()) {
-                        Thread.sleep(next - timeSystem.currentTimeInMillis());
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/util/Metronome.java]: Couldn't we simply do `Thread.sleep(abs(next - timeSystem.currentTimeInMillis())`?

**Comment by gunnarmorling** [comment]
Applied; thanks, @jpechane! Note I've added one more commit to get rid of the legacy method that didn't accept `Duration`.

**Pull request merged**